### PR TITLE
update nx, fix treeshaking, fix local webworker

### PR DESCRIPTION
**Pull request by frzyc**
A updated version of #921
With some changes to the Webpack config to 

1. Allow local web workers to work without that `import.meta` err
2. Add tree shaking
3. Add minification

Also, no more cmd windows popping up on windows.

### CODE REVIEW

File: apps/frontend/project.json [hunk 1]
```diff
@@ -59,7 +59,7 @@
           "namedChunks": false,
           "extractLicenses": true,
           "vendorChunk": false,
-          "baseHref": "/genshin-optimizer/"
+          "baseHref": "/genshin-optimizer-beta/"
```
Comment (nguyentvan7) [inline apps/frontend/project.json:62]: Probably not intended

File: apps/frontend/project.json [hunk 2]
```diff
@@ -83,7 +83,7 @@
       "dependsOn": ["^build", "^pipeline"]
     },
     "deploy":{
-      "command": "gh-pages -d dist/apps/frontend",
+      "command": "gh-pages -d dist/apps/frontend -r https://github.com/frzyc/genshin-optimizer-beta",
```
Comment (nguyentvan7) [inline apps/frontend/project.json:86]: Ditto

**Review by nguyentvan7 — CHANGES_REQUESTED** [review]

File: apps/frontend/project.json (hunk 2 above)
Comment (frzyc) [reply apps/frontend/project.json:86]: oh yeah, i was testing deployment against beta site.

File: apps/frontend/src/polyfills.ts [hunk 3]
```diff
@@ -1,7 +0,0 @@
-/**
```
Comment (nguyentvan7) [inline apps/frontend/src/polyfills.ts:1]: Does this mean we aren't using polyfills? or it's just used in a different way?

File: apps/frontend/webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+const { composePlugins, withNx } = require('@nrwl/webpack');
+const { withReact } = require('@nrwl/react');
+const TerserJSPlugin = require('terser-webpack-plugin');
+
+// Nx plugins for webpack.
+module.exports = composePlugins(withNx(), withReact(), (config) => {
+  // Update the webpack config as needed here.
+  // e.g. `config.plugins.push(new MyPlugin())`
+
+  // https://github.com/nrwl/nx/issues/10017#issuecomment-1405429851
```
Comment (nguyentvan7) [inline apps/frontend/webpack.config.js:10]: Can you leave a comment as to why we added this
```suggestion
  // https://github.com/nrwl/nx/issues/10017#issuecomment-1405429851
  // Fix to web workers not working
```

File: apps/frontend/webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  config.output.scriptType = "text/javascript"
+
+
+  // https://github.com/nrwl/nx/issues/14483#issuecomment-1424029390
+
+  // Fix 1: resolves tree-shaking issue
```
Comment (nguyentvan7) [inline apps/frontend/webpack.config.js:16]:
```suggestion
  // Fix 1: resolves tree-shaking issue caused by using custom webpack config
```

File: apps/frontend/src/polyfills.ts (hunk 3 above)
Comment (frzyc) [reply apps/frontend/src/polyfills.ts:1]: When i generated a new app, and look at their config, they dont include the polyfill anymore, so either its deprecated, or... in theory webpack does the poly fill already, and this is just a override that we are not using

**Review by nguyentvan7 — APPROVED** [review]

**Pull request merged**